libyara/scanner.c

```diff
@@ -83,7 +83,14 @@ static int _yr_scanner_scan_mem_block(
       match = match->next;
     }
 
-    index = block_data[i++] + 1;
+    __try
+	{
+		index = block_data[i++] + 1;
```

Comment:
This is not the only point where `block_data` is accessed. The call to `yr_scan_verify_match` can touch that memory too.

Reply:
Ok thanks. It should be much less critical cause I haven't seen any crash dump so far since the fix I made (local fork). Do you want me to fix or do you want to fix in your own branch?

Reply:
I just remembered that YARA already provides a try-catch (see: https://github.com/VirusTotal/yara/blob/master/libyara/scanner.c#L418). In your code, are you calling yr_scan_xxx with the `SCAN_FLAGS_NO_TRYCATCH` flag?

Reply:
Nope, I didn't even know about this flag.

Reply:
I'm afraid that exception handler isn't called in that case...

Reply:
It does, that weird. However I can confirm I haven't had any single crash dump since the fix, whereas I had at least 5-6 of them every single day. There must be something we don't see

Reply:
It's hard to "try", since I'd need to retrieve a customer's machine having that very specific driver-locking mechanism (it's tied to hardware AFAIR). 
But yeah I think you found it, EXCEPTION_CONTINUE_SEARCH will try to find a try/catch block in the frames above and probably crash if none is found. I'll let you know if I get the problem again after a new Yara update in our product.

Reply:
Thanks for looking at it :)

Reply:
Just so you know I found a crash dump failing for the same reason but this time in yr_get_pe_header. I think it's crashing when accessing the casted PE header members. It should be covered by your fix though.